rust_dev_preview/test-utils/src/macros.rs

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Macros that simplify making AWS SDK clients with a mocked request/response pair.
+
+/// Generate a single http::{Request, Response} pair. The first argument
+/// is the expression to use as the value for SdkBody::from. The second
+/// argument with the HTTP Status code and the response body.
+///
+/// To create a number of events to use for a test client, create a vec![]
+/// of many of these test_event pairs.
+#[macro_export]
+macro_rules! test_event {
+    (
+        $req:expr,
+        (
+            $status:expr,
+            $res:expr
+        )
+    ) => {{
+        (
+            http::Request::builder()
+                .body(aws_smithy_http::body::SdkBody::from($req))
+                .unwrap(),
+            http::Response::builder()
+                .status($status)
+                .body(aws_smithy_http::body::SdkBody::from($res))
+                .unwrap(),
+        )
+    }};
+}
+
+/// Create a single-shot test connection. The arguments are the same as test_event,
+/// but the expanded macro creates a TestConnection. The `TestConnection` can be
+/// provided directly to a `Client::from_conf_conn`.
+#[macro_export]
+macro_rules! single_shot {
+    (
+            $req:expr,
+            (
+                $status:expr,
+                $res:expr
+            )
+        ) => {{
+        (aws_smithy_client::test_connection::TestConnection::new(vec![
+            sdk_examples_test_utils::test_event!($req, ($status, $res)),
+        ]))
+    }};
+}
+
+/// Create a single-shot Client for `sdk_crate`. The `req` and `res` will be the
+/// body of the request and the response, respectiecly. The `status` is the HTTP
+/// status code for the response. The credentials are hardcoded test values.
+#[macro_export]
+macro_rules! single_shot_client {
+    (
+            $sdk_crate:ident,
+            $req:expr,
+            $status:expr,
+            $res:expr
+        ) => {{
+        ($sdk_crate::Client::from_conf_conn(
+            sdk_examples_test_utils::sdk_config!($sdk_crate),
+            sdk_examples_test_utils::single_shot!($req, ($status, $res)),
+        ))
+    }};
+}
```

Comment:
if you use `aws_config` here you can set the http_connector there, then you won't need another macro

Reply:
Not quite sure what you meant with this, but I've updated it to using from_conf, since from_conf_conn is removed. PTAL/Thoughts?